roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -320,10 +320,15 @@ def get_num_channels(self):
     def add_devices(imaging, nwbfile, metadata):
         # Devices
         metadata = dict_recursive_update(get_default_nwb_metadata(), metadata)
+
         # Tests if devices exist in nwbfile, if not create them from metadata
         for dev in metadata["Ophys"]["Device"]:
             if dev["name"] not in nwbfile.devices:
-                nwbfile.create_device(name=dev["name"])
+                if "description" in dev:
+                    descr = dev["description"]
+                else:
+                    descr = None
+                nwbfile.create_device(name=dev["name"], description=descr)
```

Comment:
Instead of manually adding this, I've just generalized it to `nwbfile.create_device(**device)`

Reply:
No, we were trying to instantiate a `NumpySegmentationExtractor` object with manually cut ROIs and when converting to NWB it threw an error. I agree we should have testing for the `Numpy` objects too

Reply:
What specifically is the error when writing to NWB? Is it fixed in https://github.com/catalystneuro/nwb-conversion-tools/pull/397?

Reply:
The error was a mismatch between the roi_ids and the number of available masks. I'll test if fixed on the `nwb-conv-tools` side